src/src/camp/camp.service.ts

```diff
@@ -0,0 +1,456 @@
+import { HasuraService } from '../hasura/hasura.service';
+import { HasuraService as HasuraServiceFromServices } from '../services/hasura/hasura.service';
+import { UploadFileService } from 'src/upload-file/upload-file.service';
+
+@Injectable()
+export class CampService {
+	constructor(
+		private userService: UserService,
+		private hasuraService: HasuraService,
+		private hasuraServiceFromServices: HasuraServiceFromServices,
+		private uploadFileService: UploadFileService,
+	) {}
+
+	public returnFieldsgroups = ['id', 'name','type', 'status'];
+
+	public returnFieldscamps = [
+		'kit_received',
+		'kit_was_sufficient',
+		'kit_ratings',
+		'kit_feedback',
+		'group_id',
+	];
+
+	public returnFieldsgroupUsers = ['group_id', 'id'];
+
+	async registerCamp(body: any, request: any, response: any) {
+		try {
+			let facilitator_id = request.mw_userid;
+			let learner_id = body?.learner_id;
+			let program_id = body?.program_id || 1;
+			let academic_year_id = body?.academic_year_id || 1;
+			let beneficiary_status = 'enrolled_ip_verified';
+			let createcampResponse: any;
+			let creategroupwoner: any;
+
+			let facilitator_status = await this.checkFaciltatorStatus(
+				facilitator_id,
+				program_id,
+				academic_year_id,
+			);
+			if (
+				facilitator_status?.data?.users_aggregate?.aggregate?.count == 0
+			) {
+				return response.status(401).json({
+					success: false,
+					data: {},
+					message: 'Faciltator access denied ',
+				});
+			}
+
+			// check if faciltator have more than one camps
+
+			let faciltator_camp_data = await this.checkCampInformation(
+				facilitator_id,
+			);
+			if (
+				faciltator_camp_data?.data?.camps_aggregate?.aggregate?.count >
+				1
+			) {
+				return response.status(401).json({
+					success: false,
+					data: {},
+					message: 'Camp registeration limit exceeded',
+				});
+			}
+
+			let query = `query MyQuery {
+            users(where: {_and: {program_beneficiaries: {_and: {user_id: {_in: [${learner_id}]}},status:{_eq:${beneficiary_status}}, facilitator_id: {_eq:${facilitator_id}}}}}){
+              id
+              program_beneficiaries{
+                user_id
+              }
+            }
+          }`;
+
+			const data = { query: query };
+			const res = await this.hasuraServiceFromServices.getData(data);
+			const newQdata = res?.data?.users;
+
+			// Check if learner_data is defined
+			if (
+				!newQdata ||
+				!Array.isArray(newQdata) ||
+				newQdata.length === 0
+			) {
+				return response.status(400).json({
+					success: false,
+					message: 'No learner data found or an error occurred.',
+				});
+			}
+
+			let learner_data = newQdata;
+
+			// Check if facilitator_id and learner_data have the same length
+			if (learner_id.length !== learner_data.length) {
+				return response.status(400).json({
+					success: false,
+					message:
+						'Learners do not belong to the corresponding facilitator.',
+				});
+			}
+
+			let create_group_object = {
+				name: "untitled",
```

Comment:
`name: "untitled"` change to
`name: "camp " + faciltator_camp_data?.data?.camps_aggregate?.aggregate?.count + 1`

Reply:
resolved